### ISSUE TITLE
A terminated activity could be unexpectedly restarted

### ISSUE THREAD

**Issue by tingsu** [ramack/ActivityDiary#290]
- I have searched the history bug list, and have not found a similar bug.

- The bug was found on the latest version 1.4.0 (and the master branch). The bug can be reproduced on an Android emulator 6.0 and a Google Pixel 3 XL phone.

- Bug description

A terminated activity could be unexpectedly restarted when it is followed by a undo action on any other activity.

- Video of correct functionality

Start an activity ``Cinema`` and terminate it. On the history page, we can see ``Cinema`` was correctly recorded and its state is terminated (with specific start time and end time)

![20200227_112750](https://user-images.githubusercontent.com/3156978/75446156-2d049500-5967-11ea-8842-5067d210183a.gif)

- Video of incorrect functionality

Start an activity ``Cinema`` and terminate it. Then, we start ``Sleeping`` and undo it. According to my understanding, the app now should show ``<No Activity>`` on top.  However, it shows ``Cinema``, which was already terminated. On the history page, we can also see ``Cinema`` was unexpectedly restarted (only w/ start time but wo/ end time).

![20200227_112827](https://user-images.githubusercontent.com/3156978/75446163-2fff8580-5967-11ea-8509-03f7ff51d979.gif)

**Comment by ramack** [comment]
This one I would consider a feature. I use the UNDO in case I accidentally selected a new activity, without wanting. So undoing the full action (including the termination of the previous one) seems logical. What do you think?

**Comment by tingsu** [comment]
Thanks for your explanation, Raphael.

From my end, as a user, an activity A is started, if I accidentally selected another new activity B, the ``undo`` action should cancel B and return back to A. This makes sense because I am still want to stay in activity A.  The app currently does follow this common sense. Great!

But, if I started A and then terminated A (which means I finished A), and then I accidentally selected another new activity B when I actually planned to select activity C. I hope the ``undo`` action should cancel B but should not restart A. In other words, the app should show ``<No Activity>``. Because I already terminated A, and I do not want to restart A.  Undoing the full action (including the termination of the previous one, in this case A) will force me to terminate A again before I can start C (which is annoying). Also, restarting A affects the duration of A which I selected before.

Thus, I think this could be a bug or issue that should be resolved to improve user experience. Hope my explanation makes sense :) Thanks.

**Comment by ramack** [comment]
Yes, that totally makes sense. Thanks for detailing. Clearly if the termination time of the previous activity and the start time of the new one are nit exactly the same the old one shall.not be restarted!
